src/gjk/power_to_energy.py

```diff
@@ -0,0 +1,173 @@
+from gjk.models import ReadingSql
+from gjk.models import DataChannelSql
+from gjk.config import Settings
+from sqlalchemy.orm import Session, sessionmaker
+from sqlalchemy.orm import declarative_base
+from sqlalchemy import create_engine
+from sqlalchemy import asc 
+from sqlalchemy import Column, BigInteger, Integer, String, Float
+from sqlalchemy import UniqueConstraint
+import pendulum
+import dotenv
+from datetime import datetime
+import pandas as pd
+import os
+
+# ------------------------------------------
+# User inputs
+# ------------------------------------------
+
+timezone = 'America/New_York'
+start= pendulum.datetime(2024, 1, 1, 0, 0, tz=timezone)
+end = pendulum.datetime(2024, 2, 2, 20, 0, tz=timezone)
+
+# Record 0 Wh values in the energy table
+record_zeros = False
+
+# ------------------------------------------
+# Import readings data
+# ------------------------------------------
+
+dotenv.load_dotenv()
+engine = create_engine(os.getenv('GJK_DB_URL'))
+Session = sessionmaker(bind=engine)
+session = Session()
+
+saved_sql_channels = session.query(DataChannelSql).all()
+saved_channel_ids = [channel.id for channel in saved_sql_channels]
+
+# Convert to Unix timestamp in milliseconds
+start_ms = int(start.add(hours=-1).timestamp() * 1000)
+end_ms = int(end.timestamp() * 1000)
+
+readings = session.query(ReadingSql).filter(
+    ReadingSql.time_ms >= start_ms,
+    ReadingSql.time_ms < end_ms,
+).order_by(asc(ReadingSql.time_ms)).all()
+
+# ------------------------------------------
+# Split the data by channel
+# ------------------------------------------
+
+power_channels = ['hp-idu', 'hp-odu', 'dist-pump', 'primary-pump', 'store-pump']
+
+power_data = {}
+time_data = {}
+for p in power_channels:
+    pwr_values = []
+    pwr_times = []
+    for r in readings:
+        if p in r.data_channel.name:
+            pwr_values.append(r.value)
+            pwr_times.append(r.time_ms)
+    power_data[f'{p}'] = pwr_values
+    time_data[f'{p}'] = pwr_times
+
+# ------------------------------------------
+# For every hour, convert to energy
+# ------------------------------------------
+
+energy_results = {}
+
+first_hour = start.add(hours=1).in_tz('UTC').naive()
+last_hour = end.in_tz('UTC').naive()
+num_hours = int((last_hour-first_hour).total_hours())
+hour_data = [first_hour.add(hours=x) for x in range(num_hours)]
+
+for p in power_channels:
+    
+    energy_results[f'{p}'] = []
+    time_data_hours = [datetime.fromtimestamp(x/1000).replace(minute=0, second=0, microsecond=0) for x in time_data[f'{p}']]
```

Comment:
datetime.fromtimestamp gives local time in a timezone naive format. Use pendulum.from_timestamp instead:

>>> datetime.datetime.fromtimestamp(reading.time_ms / 1000)
datetime.datetime(2024, 2, 1, 0, 0, 57, 539000)
>>> pendulum.from_timestamp(reading.time_ms / 1000)
DateTime(2024, 2, 1, 5, 0, 57, 539000, tzinfo=Timezone('UTC'))